resolving/npm-resolver/src/parsePref.ts

```diff
@@ -49,3 +50,24 @@ export function parsePref (
   }
   return null
 }
+
+export function parseJsrPref (
+  pref: string,
+  alias: string | undefined,
+  defaultTag: string,
+  registry: string
+): RegistryPackageSpec {
+  let spec = parsePref(pref, alias, defaultTag, registry)
+  if (!spec) {
+    spec = parsePref(`npm:${pref}`, alias, defaultTag, registry)
```

Comment:
Move the related code to a new function instead of prepending `npm:`.

Also, in which case will the first invocation of parsePref return undefined?

Reply:
> Also, in which case will the first invocation of parsePref return undefined?

When `pref` is not a valid npm spec, such as when it starts with `jsr:`.

Reply:
But I see that you remove the jsr prefix before passing pref to parseJsrPref.

I think it would make more sense to remove the prefix in parseJsrPref and return undefined if it is not a jsr spec. In that case you can move parseJsrPref to the top of the tryResolveFromJsr function.

Reply:
* https://github.com/pnpm/pnpm/pull/9358/commits/33596ac60374ea9260571dd56ce091f13650bacd
* https://github.com/pnpm/pnpm/pull/9358/commits/a511dbb1374c0138afe751e6f43abd9c4527b73b